app/models/host/discovered.rb

```diff
@@ -9,8 +9,6 @@ class Host::Discovered < ::Host::Base
   belongs_to :hostgroup
   has_one    :discovery_attribute_set, :foreign_key => :host_id, :dependent => :destroy
 
-  validates :mac, :uniqueness => true, :mac_address => true, :presence => true
-  validates :ip, :format => {:with => Net::Validations::IP_REGEXP}, :uniqueness => true
```

Comment:
it's not great to run the validations twice but if that's the only problem maybe keep this in for now if you are trying to make this compatible to 1.7?